### ISSUE TITLE
False positive for `no-implicit-coercion` rule

### ISSUE THREAD

**Issue by sindresorhus** [eslint/eslint#4410]
```
~/dev
❯ eslint --version                                                                         
v1.9.0
                                                                                                                                               
~/dev
❯ echo 'functionReturningAnInteger() * 1' | eslint --stdin --rule='no-implicit-coercion: 2'

<text>
  1:1  error  use `Number(functionReturningAnInteger())` instead  no-implicit-coercion

✖ 1 problem (1 error, 0 warnings)
```

It shouldn't require me to explicitly coerce function calls unless it can determine that the function doesn't return a number, which I'm pretty sure ESLint can't do.

**Comment by michaelficarra** [comment]
Why are you multiplying by 1?

**Comment by sindresorhus** [comment]
Yeah, that makes no sense :p

Was just copied from a PR I got: https://github.com/sindresorhus/caprine/pull/30/files#diff-168726dbe96b3ce427e7fedce31bb0bcR81

I should have looked at it closer. Sorry for the noise.
